hazelcast/src/hazelcast/client/xml_impl.cpp

```diff
@@ -0,0 +1,1317 @@
+namespace config {
+
+abstract_dom_config_processor::abstract_dom_config_processor(bool dom_level_3)
+{
+    this->dom_level_3 = dom_level_3;
+    this->strict = true;
+}
+abstract_dom_config_processor::abstract_dom_config_processor(bool dom_level_3,
+                                                             bool strict)
+{
+    this->dom_level_3 = dom_level_3;
+    this->strict = strict;
+}
+
+bool
+abstract_dom_config_processor::matches(const std::string& config1,
+                                       const std::string& config2)
+{
+    return config1 == config2;
+}
+std::string
+abstract_dom_config_processor::get_attribute(boost::property_tree::ptree node,
+                                             const std::string& attribute)
+{
+    return node.get_child("<xmlattr>." + attribute).data();
+}
+bool
+abstract_dom_config_processor::get_bool_value(std::string value)
+{
+    boost::algorithm::to_lower(value);
+    if (value == "true") {
+        return true;
+    } else if (value == "false") {
+        return false;
+    }
+    return false;
+}
+int
+abstract_dom_config_processor::get_integer_value(
+  const std::string& parameter_name,
+  const std::string& value)
+{
+    try {
+        return std::stoi(value);
+    } catch (const std::exception& e) {
+        throw hazelcast::client::exception::invalid_configuration(
+          "Invalid integer value for parameter " + parameter_name + ": " +
+          value);
+    }
+}
+long
+abstract_dom_config_processor::get_long_value(const std::string& parameter_name,
+                                              const std::string& value)
+{
+    try {
+        return std::stol(value);
+    } catch (const std::exception& e) {
+        throw hazelcast::client::exception::invalid_configuration(
+          "Invalid long value for parameter " + parameter_name + ": " + value);
+    }
+}
+double
+abstract_dom_config_processor::get_double_value(
+  const std::string& parameter_name,
+  const std::string& value)
+{
+    try {
+        return std::stod(value);
+    } catch (const std::exception& e) {
+        throw hazelcast::client::exception::invalid_configuration(
+          "Invalid double value for parameter " + parameter_name + ": " +
+          value);
+    }
+}
+
+
+hazelcast::client::serialization_config
+abstract_dom_config_processor::parse_serialization(
+  boost::property_tree::ptree node)
+{
+    hazelcast::client::serialization_config serialization_config ;
+    for (auto& child : node) {
+        std::string name = child.first;
+        if (matches("portable-version", name)) {
+            std::string value = child.second.data();
+            serialization_config.set_portable_version(
+              get_integer_value(child.first, value));
+        } else if (matches("use-native-byte-order", name)) {
+            serialization_config.set_byte_order(boost::endian::order::native);
+        } else if (matches("byte-order", name)) {
+            std::string value = child.second.data();
```

Comment:
If the value is neither 'BIG_ENDIAN' nor 'LITTLE_ENDIAN' it should be defaulted to 'BIG_ENDIAN'